Source/WebGPU/WebGPU/PresentationContextIOSurface.mm

```diff
@@ -129,6 +129,35 @@
         if (iosurface.height != static_cast<NSInteger>(height) || iosurface.width != static_cast<NSInteger>(width))
             return device.generateAValidationError("Invalid surface size"_s);
     }
+
+    if (!allowedFormat(descriptor.format)) {
```

Comment:
validation should occur before texture creation, we were creating it later due to our implementation previously not being able to handle a mismatch in textures and IOSurfaces. That is now resolved.